### PR TITLE
Add Cronet to main build

### PR DESCRIPTION
**Pull request by ejona86**
~~This is a draft as the CI still needs to be adjusted and at will cause merge conflicts with #6130. I'm sending this out now so that it is more clear what #6131 permits. Note that this includes #6131, so much of it is noise; just look at the last commit.~~

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -208,6 +212,22 @@ subprojects {
         }
     }
 
+        if (rootProject.properties.get('errorProne', true)) {
```
Comment (voidzcy) [inline build.gradle]: The indentation of this block may need to be fixed.

**Comment by ejona86** [comment]
We should do this same thing for grpc-android and grpc-android-interop-testing (potentially removing the useless parent project that exists there today; "inline" `app`). But I was happy just doing one for now and seeing how well it went.

File: COMPILING.md [hunk 2]
```diff
@@ -11,6 +11,11 @@ developers don't have C compilers installed and don't need to run or modify the
 codegen, the build can skip it. To skip, create the file
 `<project-root>/gradle.properties` and add `skipCodegen=true`.
 
+Some parts of grpc-java depend on Android. Since many Java developers don't have
+the Android SDK installed and don't need to run or modify the Android
```
Comment (sanjaypujare) [inline COMPILING.md]: Are there instructions to install the Android SDK in case someone doesn't want to "skipAndroid"?

Comment (ejona86) [reply COMPILING.md]: It is basically download the SDK and set $ANDROID_HOME. But there's some complications that make it hard to be concise. I think I'll let @dapengzhang0 or @voidzcy add a documentation section later in this document, but after this is merged.

(Among the problems: ANDROID_HOME is actually deprecated now in favor of ANDROID_SDK_HOME, but we'd need to upgrade to a newer gradle plugin for ANDROID_SDK _HOME to work)

Comment (voidzcy) [reply COMPILING.md]: Sure. It's in #6593.

**Pull request merged**